### ISSUE TITLE
Is there a way to config min_sample_weight_in_leaf during model training for Binary Classification?

### ISSUE THREAD

**Issue by xuyannus** [microsoft/LightGBM#2438]
Question:
I am using LGBMClassifier and sample_weight to train a binary classification model. I have a problem to deal with overfitting when sample_weight is small.

It seems to me that min_data_in_leaf does not consider sample_weight. So, the question is: how to stop the tree to split if the sum of sample_weights in the leaf is small than a value (e.g. 100)

Could anyone please help?

Example Code: 
------
    inputs_df = pd.DataFrame([[1, 5, 400],
                              [1, 5, 600],
                              [1, 2, 50],
                              [1, 2, 50],
                              [2, 3, 10],
                              [2, 3, 40],
                              [3, 4, 2],
                              [3, 4, 1]], columns=['feature1', 'feature2', 'weights'])
    outputs = [1, 0, 1, 0, 1, 0, 1, 0]
    features = ['feature1', 'feature2']

    hyper_params = {"min_data_in_leaf": 0, "max_depth": -1, "objective": "binary", "boosting_type": "gbdt", "n_estimators": 10, "lambda_l2": 0.0, "learning_rate": 1.0, "num_leaves": 31}

    model = lgb.LGBMClassifier(**hyper_params)
    model.fit(inputs_df[features],
              outputs,
              sample_weight=inputs_df['weights'].values,
              eval_metric='binary_logloss',
              feature_name=features,
              categorical_feature=features)

------

## Environment info

Operating System: MacBookPro

CPU/GPU model: CPU

C++/Python/R version: Python 3.6.8

LightGBM version or commit hash: 2.2.3

**Comment by guolinke** [comment]
you can use `min_child_weight`

**Comment by xuyannus** [comment]
@guolinke the document says min_child_weight  is the same with min_sum_hessian. I am confused on what exactly does it do. any suggestions?

**Comment by guolinke** [comment]
It is similar to the min_data, but use the second order gradients, and consider the sample weight. In regression, it equals to min_data.

**Comment by xuyannus** [comment]
thanks, @guolinke back to the question: how to stop the tree to split if the sum of sample_weights in the leaf is small than a value (e.g. 100). I am not sure how should I set up min_child_weight in a binary classification problem. should I use: min_child_weight = 100, or something else. 

PS: thanks a lot for your fast reply,.

**Comment by guolinke** [comment]
The second order gradient is p(1-p) in binary objective. So it is hard to have a accurate value for your need. You can base on the distribution if your data, and estimate a reasonable value
